src/reducers/project-state.js

```diff
@@ -435,10 +435,21 @@ const onLoadedProject = (loadingState, canSave, success) => {
         default:
             return;
         }
+    } else {
```

Comment:
This code seems to be getting more confusing because of the combination of `if` and `switch`. I realize that's the way the code was already written, but for example, I think this `else` is not needed since there's a `default: return` above. However, that fact is not immediately obvious/easy to reason about. 

Do you have any thoughts on how we could simplify this?

Reply:
Thanks for pointing this out. I tried using one switch statement instead, and I think it's better:

```
    switch (loadingState) {
    case LoadingState.LOADING_VM_WITH_ID:
        if (success) {
            return {type: DONE_LOADING_VM_WITH_ID};
        }
        // failed to load project; just keep showing current project
        return {type: RETURN_TO_SHOWING};
    case LoadingState.LOADING_VM_FILE_UPLOAD:
        if (success) {
            if (canSave) {
                return {type: DONE_LOADING_VM_TO_SAVE};
            }
            return {type: DONE_LOADING_VM_WITHOUT_ID};
        }
        // failed to load project; just keep showing current project
        return {type: RETURN_TO_SHOWING};
    case LoadingState.LOADING_VM_NEW_DEFAULT:
        if (success) {
            return {type: DONE_LOADING_VM_WITHOUT_ID};
        }
        // failed to load default project; show error
        return {type: START_ERROR};
    default:
        return;
    }
```

Reply:
This feels much better/easier to reason about!